src/test/java/seedu/duke/commands/ShowRateCommandTest.java

```diff
@@ -1,17 +1,24 @@
 package seedu.duke.commands;
 
-import org.junit.jupiter.api.Test;
 import seedu.duke.ui.Ui;
 import seedu.duke.AccountList;
+import seedu.duke.Forex;
+import seedu.duke.Currency;
+import seedu.duke.exceptions.InvalidNumberException;
+
+import java.math.BigDecimal;
 
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.fail;
+import org.junit.jupiter.api.Test;
 
 public class ShowRateCommandTest {
 
     @Test
     public void testExecute_correctSyntax_shouldNotException() {
         try {
+            Forex.initializeRates();
```

Comment:
Hmm initializeRates seem to be more for integration tests and we might have to use mocks for it. Might have to check with @Bawfen whether this is ok

Reply:
I called initializeRates() for the first test only, because the static array of rates in Forex will then be initialized for the other tests. After we merge this PR I am going to improve testing for show-rates by extracting the parsing to a new method, so I can work on it then.

Reply:
Also on this note, the API is also being tested through these tests, which I think is important to do. Using mocks may not reveal errors with the API, such as when our monthly calls expire. Replacing the API key is very easy, so I am in favor of using the API for testing, but happy to change it.